inference/huggingface/zero_inference/run_model.py

```diff
@@ -155,32 +154,25 @@ def run_generation(
     quant_group_size,
     pin_kv_cache,
     async_kv_offload,
+    loops,
 ):
     # Load tokenizer
-    config = get_model_config(model_name)
-    return_token_type_ids = True 
-    padding_side = "left" if config.model_type in ["opt"] else "right"
+    config = get_model_config(model_name)    
 
     if config.model_type == "opt":
         tokenizer = AutoTokenizer.from_pretrained(
             model_name.replace("175b", "66b"), 
-            return_token_type_ids=return_token_type_ids,
-            padding_side=padding_side
+            padding_side="left" 
```

Comment:
Perhaps this part can be refactored a bit to be tokenizer = get_tokenizer(config) and move all model-specific configurations (e.g., opt specific hack and pad token assignment) into the get_tokenizer function. This helps to make the code in the main generation function cleaner.

Reply:
Done!